### ISSUE TITLE
Adjust vertical metrics of Hind

### ISSUE THREAD

**Issue by Mel-Roth-AI** [google/fonts#564]
[Hind](https://fonts.google.com/specimen/Hind?selection.family=Hind) vertically aligns too high:

![hind-vertical-metrics](https://cloud.githubusercontent.com/assets/20308275/21655348/88982180-d26e-11e6-9e99-93503d0e5bf3.png)

![hind-vertical-metrics-button](https://cloud.githubusercontent.com/assets/20308275/21655698/f163c70e-d26f-11e6-94c0-f49a39921896.png)

![hind-vertical-metrics-table](https://cloud.githubusercontent.com/assets/20308275/21655708/f6db0ba2-d26f-11e6-8698-fe286b789e72.png)

**Comment by m4rc1e** [comment]
Can you provide more information? A comparison should be sufficient.

I can't see a problem. If you feel the vertical metrics are too loose, it may be because it has Hindi. Hindi has above and below marks on consonants which can be much taller/shorter than Latin glyphs.

**Comment by davelab6** [comment]
I agree, the metrics are like this because of the Devangari

**Comment by Mel-Roth-AI** [comment]
I'm not quite sure this is the comparison you're looking for, but a font that supports Devanagari with good vertical metrics is https://fonts.google.com/specimen/Noto+Sans

**Comment by asadkn** [comment]
This is a common issue with many Devanagari fonts and popular ones such as Poppins. And yes, Nunito Sans and almost every premium font does a much better job at it.

**Comment by m4rc1e** [comment]
@lianghai changed the vertical metrics #1099 

![screen shot 2017-07-25 at 11 49 19](https://user-images.githubusercontent.com/7525512/28568940-85925c84-712f-11e7-9eaa-54d79c982e38.png)

I'll keep this open for the time being.

**Comment by davelab6** [comment]
I suspect this should become a fontbakery issue, but moving this to the font development milestone track for now and leaving it open.

**Comment by laerm0** [comment]
Well, the Hind / Noto Sans comparison is a good one to make as the Devanagari is the same in both. Here's where the difference comes in:

 
![screen region 2017-11-13 at 15 29 53 copy](https://user-images.githubusercontent.com/10780161/32748129-47b7e012-c889-11e7-82d8-71f56f0140f0.png)

_Hind above, Noto Sans below_

The ascender values in Hind are higher. (It's also set to UseTypoMetrics.) I'm going to see if changing Hind's ascender values to Noto's fixes this issue.

**Comment by laerm0** [comment]
I played around with various combinations of metrics, including measuring and recalculating them all from scratch according to MS standards and Glyphapp's recommendations for webfonts and was able to make nothing better than this:

![screen region 2017-11-14 at 12 12 23 copy](https://user-images.githubusercontent.com/10780161/32794064-d64fed4a-c935-11e7-8ce4-127b37083734.png)

_Hind on top, Noto below, altered Hind last_

I could literally make the metrics of Hind equal Noto and still get lots of space on top. That space is largely due to Hind having an x-height and cap height lower than Noto.

Thoughts?

**Comment by davelab6** [comment]
> the Devanagari is the same in both
> ...
> _Hind above, Noto Sans below_

I'm confused; I only see the Noto design in your images. 

https://fonts.google.com/selection?query=noto&selection.family=Hind|Noto+Sans&subset=devanagari

`आकाश में बादल नहीं थे और उसका रंग गहरा नीला था ।`

<img width="675" alt="screen shot 2017-11-14 at 15 29 37" src="https://user-images.githubusercontent.com/261579/32803169-aa2d50d4-c950-11e7-9e27-42611c1b5581.png">

**Comment by laerm0** [comment]
Whoa, weird: all of my images were from Word. They're definitely the same there!

**Comment by davelab6** [comment]
> Whoa, weird: all of my images were from Word. They're definitely the same there!

I am curious why you would test on MS Word, instead of a browser (eg Chrome) first? :) 

I think the OP issue is that when you highlight Noto, its selection bounds seem "normal" but in Hind they seem "deep":

<img width="254" alt="google chromescreensnapz005" src="https://user-images.githubusercontent.com/261579/32803425-6fbb945a-c951-11e7-9e20-1aa0f0de7fc5.png">
<img width="254" alt="google chromescreensnapz005" src="https://user-images.githubusercontent.com/261579/32803429-71354344-c951-11e7-9f9d-6b3f9e918c25.png">

**Comment by laerm0** [comment]
I guess because it was easiest to make a test doc in Word and because usually it's more strict with vertical metrics than browsers.

Yeah, I'll need to work on this more. The current fonts also clip in-browser. Hmm.

![screen region 2017-11-14 at 15 39 27](https://user-images.githubusercontent.com/10780161/32803612-1565155c-c952-11e7-9279-133110e7b5f8.png)

**Comment by laerm0** [comment]
At any rate, I do suspect that the descent values are the culprit as the descenders are set for 546 in Hind and nearly half that in Noto.

**Comment by laerm0** [comment]
Windows does display the fonts properly, by the way. Guess that is a bug in OS X Word?

![screen region 2017-11-14 at 15 44 30](https://user-images.githubusercontent.com/10780161/32803850-d8a514ae-c952-11e7-9170-89945da9890b.png)

**Comment by davelab6** [comment]
Please post your test string here also :)

**Comment by davelab6** [comment]
If macOS Word can't render Hind's devanagari at all, that is a separate bug that needs to be filed ;)

**Comment by davelab6** [comment]
> At any rate, I do suspect that the descent values are the culprit as the descenders are set for 546 in Hind and nearly half that in Noto.

Right - the question is, how can we improve the text selection display in multiscript families (like Hind) where v metrics make highlight very tall? Is there a better V Metrics setting that will (a) prevent clipping and (b) allow for a good selection display?

**Comment by laerm0** [comment]
Test string: Lorem ipsum dolor sit amet ऋषियों को सताने वाले दुष्ट राक्षसों

> If macOS Word can't render Hind's devanagari at all, that is a separate bug that needs to be filed ;)

Indeed!

> Right - the question is, how can we improve the text selection display in multiscript families (like Hind) where v metrics make highlight very tall? Is there a better V Metrics setting that will (a) prevent clipping and (b) allow for a good selection display?

Yep, this will be my project now. I'm going to look at some other multiscript families and get a theory going for us.

**Comment by davelab6** [comment]
Sounds good - please map out time for this in the tracker, I'm not sure
this is the highest priority thing to work on, if its more than half a day
we should plan for it :)

**Comment by laerm0** [comment]
Yes, I will take a look at that later tonight. I think this is the kind of thing that will help in multiple families, though, so perhaps it's worth a higher priority. (I used to have a script that checked to see if things clipped. I think it was AFDKO thing...)

**Comment by davelab6** [comment]
Yes, a script is the easy hard - this can be resolved to some simple
arithmetic checks - rather, its knowing what the ranges of values and
combinations of values are which is the tricky part :)

**Comment by laerm0** [comment]
I'm going to spend a couple days on this. This morning I have figured out which Devanagari fonts are good and which are bad. The good ones are really only eight of them:

Noto Sans
Poppins
Rozha One
Rajdhani
Yantramanav
Teko
Amiko
Sarpanch

Those all have reasonable metrics and no overlap of lines. I don't think it's coincidental five of the top eight fonts on GF have good metrics.

The bad list is much longer:

Hind (very bad)
Khula (very bad)
Ranga (very bad)
Kalam
Khand
Glegoo (very bad)
Pragati Narrow (very bad)
Sarala
Martel
Sumana (very bad)
Jaldi
Cambay
Palanquin (very bad)
Halant
Kurale
Karma
Martel Sans (very bad)
Baloo
Biryani
Palanquin Dark (very bad)
Eczar
Vesper Libre (very bad)
Kadwa (very bad)
Mukta
Rhodium Libre (very bad)
Dekko
Arya (very bad)
Laila
Amita (very bad)
Sahitya
Tillana
Modak
Yatra One
Inknut Antiqua (very bad)
Sura
Asar

These all have lines that overlap each other because linegaps are zero and ascender and descender values are more than the UPM (2.5× worse or so in the ones I labeled as very bad).

If I am going to fix one (i.e. Hind), I think it makes sense to fix them all, once I figure out a standard. Let me check and see if we have one...

[https://github.com/googlefonts/gf-docs/blob/master/ProjectChecklist.md]

>In July 2016, Khaled Hosny and Kalapi Gajjar proposed two new vertical metrics schemes which are now recommended (discussion.)

>Khaled's scheme is recommended for projects that followed Raph's 2011 recommendation; in most cases it doesn't create any visible repositioning in any browsers, but testing this is required for each family that is updated.

>TODO describe Khaled's scheme

>Kalapi's scheme is recommended for new projects.

>TODO describe Kalapi's scheme

🤣
Okay, I guess I'll read the thread and write up the standards discussed and then add them to the ProjectChecklist.md doc.

**Comment by laerm0** [comment]
Oh wait, we do have a [vertical metrics doc](https://github.com/googlefonts/gf-docs/blob/master/VerticalMetrics.md). I'll combine working on this issue with #1284.

**Comment by laerm0** [comment]
Okay, so following the metrics doc, here's what Hind looks like now.

![screen region 2017-11-15 at 15 02 04](https://user-images.githubusercontent.com/10780161/32857661-91350a2e-ca16-11e7-9d83-da862fd78e36.png)

I'm calling this a good improvement. We've got a few pixels on either side of the yMax and yMin so I don't think it'll clip. I'm going to go through a few of the other troubled fonts and see how they look after this.

**Comment by laerm0** [comment]
Okay, I've tested a few of the other iffy ones and this seems to be a great improvement for all of them. 

For example, here's Jaldi before:

![screen region 2017-11-15 at 16 09 28](https://user-images.githubusercontent.com/10780161/32860359-63f68142-ca1f-11e7-8db1-8c9f9008d7d8.png)


After:

![screen region 2017-11-17 at 10 27 06](https://user-images.githubusercontent.com/10780161/32954722-e3466cb0-cb81-11e7-8379-673a3b8c1c37.png)

@davelab6 Should I go through all the Indian scripts and update vertical metrics according to GF spec?

**Comment by laerm0** [comment]
Hey @davelab6, what do you think? Should I clean up all the iffy fonts I found?

**Comment by asadkn** [comment]
Any update on this? 

To add to the growing list - another one with awful vertical alignment: "Arima Madurai (very bad)"

**Comment by laerm0** [comment]
Thanks, @asadkn. Fixing all of these is on my to-do list in the next few months.

**Comment by billyudi** [comment]
Anyword on fixing Cambay?

**Comment by chrissimpkins** [comment]
>Oh wait, we do have a [vertical metrics doc](https://github.com/googlefonts/gf-docs/blob/master/VerticalMetrics.md). I'll combine working on this issue with #1284.

@laerm0 is this vertical metrics doc for Indic scripts still available somewhere? It links to 404 page now.

**Comment by laerm0** [comment]
@chrissimpkins Hmm, bummer. I bet someone with GitHub-fu can dig it out of history, though.

**Comment by chrissimpkins** [comment]
@laerm0 OK, thanks!

**Comment by laerm0** [comment]
Well, I'm not suggesting _I_ know how to do it... 😆

**Comment by asadkn** [comment]
The previously linked vertical metrics doc is probably here now: https://github.com/googlefonts/gf-docs/tree/master/VerticalMetrics

**Comment by chrissimpkins** [comment]
@asadkn ty!
